### PR TITLE
[opt](http) enable auth token with BE http request

### PR DESCRIPTION
**Pull request by morningman**
### What problem does this PR solve?
Related PR: #39577

Problem Summary:
In #39577, we added the auth check for all HTTP API on FE side.
But it introduced an issue that when `enable_all_http_auth`, the internal http request
will fail due to lack of authentication info.
For example, when cloning replica from one BE to another, it use HTTP API.

This PR mainly changes:

1. Unify the token generation and checking logic

	Move `TokenManager` from `load` package to `Env`, as a global mgr.

	It is responsible for generating tokens at fix interval.
	And the token will be sent to BE via heartbeat.

	BE will save last 2 tokens, and use the latest token in HTTP request.

	All HTTP request sent by BE will add a header `Auth-Token`,
	and BE's HTTP server will check if this token in header is same as token
	from FE heartbeat.

2. Add a new class `ClusterInfo` on BE side to replace `TMasterInfo`.

	`TMasterInfo` is a thrift object used to save master info and pass them
	from FE to BE via heartbeat.
	So it should only be a message payload, we should get info from it and
	save it in another structure: `ClusterInfo`.

### Check List (For Committer)

- Test 

    - [ ] Regression test
    - [x] Unit Test
    - [x] Manual test
        I created a cluster with 2 BE, set `enable_all_http_auth=true`. And create a table with 1 replica,
        and then modify the replica num to 2. The clone task run success.
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No colde files have been changed.
        - [ ] Other reason 

- Behavior changed:

    - [x] No.
    - [ ] Yes. 

- Does this need documentation?

    - [x] No.
    - [ ] Yes. 

- Release note

    
    
    None

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by morningman** [comment]
run buildall

**Comment by morningman** [comment]
run buildall

**Comment by morningman** [comment]
run buildall

**Comment by gavinchou** [comment]
how do we get it tested?

**Comment by morningman** [comment]
run buildall

File: gensrc/thrift/HeartbeatService.thrift [hunk 1]
```diff
@@ -43,6 +43,8 @@ struct TMasterInfo {
     11: optional string cloud_unique_id;
     // See configuration item Config.java rehash_tablet_after_be_dead_seconds for meaning
     12: optional i64 tablet_report_inactive_duration_ms;
+    13: optional string last_auth_token;
```
[the remark this answers is not in the document]
Comment (morningman) [reply gensrc/thrift/HeartbeatService.thrift]: removed

File: be/src/runtime/cluster_info.h [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    TNetworkAddress master_fe_addr;
+    // Master FE http_port
+    int32_t master_fe_http_port = 0;
+    // Unique cluster token
+    std::string token = "";
+    // Backend ID
+    int64_t backend_id = 0;
+
+    // Auth token for internal authentication
+    std::string curr_auth_token = "";
```
[the remark this answers is not in the document]
Comment (morningman) [reply be/src/runtime/cluster_info.h:44]: added comment

File: be/src/runtime/cluster_info.h [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+#include <gen_cpp/Types_types.h>
+
+#include <string>
+
+namespace doris {
+
+// This class is used to save the cluster info
+// like cluster id, epoch, cloud_unique_id, etc.
+// These info are usually in heartbeat from Master FE.
+class ClusterInfo {
```
[the remark this answers is not in the document]
Comment (morningman) [reply be/src/runtime/cluster_info.h:29]: class and struct is same?

**Comment by morningman** [comment]
> how do we get it tested?

already described in PR description. But we also need test it in a multi node env

**Review by hubgeter — APPROVED** [review]

**Review by kaka11chen — APPROVED** [review]
LGTM

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**